### ISSUE TITLE
Gamma and brightness settings in a game don't affect the recording

### ISSUE THREAD

**Issue by MikkoMMM** [MaartenBaert/ssr#339]
I was recording a game called The Dark Mod. However, I noticed that the recording of it was too dark. Then I tested what effect the game's own gamma and brightness sliders have and they didn't have any effect on the recording. What the? My intuition would tell me SimpleScreenRecorder would record the screen exactly as I see it, but apparently it doesn't. For now I can increase the gamma in post-production, but I would rather have it be bright enough from the get-go. It probably decreases the amount of detail the way I have to do it now.

Here are some recording options I used that could be relevant:
* Record the entire screen (non-OpenGL)
* Codec: H.264

I'm using SimpleScreenRecorder 0.3.3 on Arch Linux. My graphics card is Asus GTX780-DC20C-3GD5 3GB.

Here's what SSR says when I start it:
SimpleScreenRecorder 0.3.3
Compiled with GCC 4.9.2
Qt: header 4.8.6, lib 4.8.6
libavformat: header 56.15.102, lib 56.25.101
libavcodec: header 56.13.100, lib 56.26.100
libavutil: header 54.15.100, lib 54.20.100
libswscale: header 3.1.101, lib 3.1.101
[CPUFeatures::Detect] CPU features: mmx sse sse2 sse3 ssse3 sse4_1 sse4_2 avx avx2 bmi1 bmi2
[ALSAInput::GetSourceList] Generating source list ...
[ALSAInput::GetSourceList] Found card: [hw:0] HDA Intel PCH
[ALSAInput::GetSourceList] Found device: [hw:0,0] HDA Intel PCH: ALC1150 Analog
[ALSAInput::GetSourceList] Found device: [hw:0,2] HDA Intel PCH: ALC1150 Alt Analog
[ALSAInput::GetSourceList] Found card: [hw:1] HDA NVidia
[PulseAudioInput::GetSourceList] Generating source list ...
[SourceNamesCallback] Found source: [alsa_output.pci-0000_00_1b.0.analog-stereo.monitor] Monitor of Built-in Audio Analog Stereo

**Comment by StripedMonkey** [comment]
Did you try to stop and restart the recording?

**Comment by MikkoMMM** [comment]
Yes, I stopped and restarted the recording two times and the problem
persisted in each recording.

2015-06-08 17:12 GMT+03:00 Noah <notifications@github.com>:

> Did you try to stop and restart the recording?
>

**Comment by MaartenBaert** [comment]
SSR records the raw RGB data that is stored in the graphics card. Some games use the gamma correction settings of the video card to get certain effects, but this correction is applied only when the data is sent out of the GPU to your display, so SSR can't see this. I would have to manually figure out what the gamma settings are and re-apply this to the image in SSR, which is tricky and will have a significant performance impact.

The problem is, many users actually use gamma settings to correct the color of their display. In fact this is how gamma correction is meant to be used, the games are just abusing it for something else. There are also some programs (e.g. redshift) that adjust the gamma curves of your screen depending on the time of day. In those cases you definitely don't want to see the effect of gamma correction in the final video.

I suppose I could add a checkbox to enable gamma correction in SSR, but it's not high on my todo list considering how few games actually use this. So for now, do the correction in post-production (this does increase noise somewhat, but that's unavoidable with gamma correction).

**Comment by tenleftfingers** [comment]
I'll post here because I think it's the same issue. But my recordings of Ubuntu desktop are way too bright and I don't know how to retrieve the lost colour which has been washed out.

I'm using default settings on 14.04 - any suggestions? I do use RedShift but quit the app before recording. I've attached a screengrab.
![ssr](https://cloud.githubusercontent.com/assets/1757768/10947234/c8925404-831e-11e5-84d5-e96ac6d41ce2.png)

**Comment by HybridDog** [comment]
As far as I can see, videos encoded by simplescreenrecorder use the bt.1886 gamma. Under the assumption that screens use sRGB by default, does this gamma lead to a video file with a slightly wrong brightness?
